src/Assert.php

```diff
@@ -401,7 +407,44 @@ public static function isInstanceOfAny($value, array $classes, $message = '')
         }
 
         static::reportInvalidArgument(sprintf(
-            $message ?: 'Expected an instance of any of %2$s. Got: %s',
+            $message ?: 'Expected an instance of this class or to this class among his parents of any of %2$s. Got: %s',
+            static::typeToString($value),
+            implode(', ', array_map(array('static', 'valueToString'), $classes))
+        ));
+    }
+
+    public static function isAOf($value, $class, $message = '')
+    {
+        if (!is_string($class) || !is_a($value, $class, is_string($value))) {
+            static::reportInvalidArgument(sprintf(
+                $message ?: 'Expected an instance of this class or to this class among his parents %2$s. Got: %s',
+                static::typeToString($value),
+                $class
+            ));
+        }
+    }
+
+    public static function notAOf($value, $class, $message = '')
+    {
+        if (!is_string($class) || is_a($value, $class, is_string($value))) {
+            static::reportInvalidArgument(sprintf(
+                $message ?: 'Expected an instance of this class or to this class among his parents other than %2$s. Got: %s',
+                static::typeToString($value),
+                $class
+            ));
+        }
+    }
+
+    public static function isAOfAny($value, array $classes, $message = '')
+    {
+        foreach ($classes as $class) {
+            if (!is_string($class) || is_a($value, $class, is_string($value))) {
```

Comment:
I would have the assumption if no string passed for `$class` an exception would be thrown. If you enabled `declare(strict_types=1);` and calling `is_a()` with a class variable contains an integer php would thrown a type error. I talked with @Nyholm and he agreed if `$class` is not a string a an invalid argument should be reported.